src/SweetPoll.elm

```diff
@@ -66,18 +66,29 @@ init config =
         , config = config
         }
         |> runPoll
-        |> -- lastData will always be Nothing here, so strip it out to keep init's type signature nicer
-           (\( model, _, cmd ) -> ( model, cmd ))
+        |> -- lastData and error will always be Nothing here,
+           -- so strip it out to keep init's type signature nicer
+           (\result -> ( result.sweetPollModel, result.cmd ))
 
 
-{-| The SweetPoll StartApp-style update function
+{-|
+
+ - sweetPollModel: the new state of the SweetPoll
+ - currentData: the most recent data received by the SweetPoll
+ - error: any new error occurring in the current update cycle
+ - cmd: a Cmd to keep the SweetPoll running
+-}
+type alias UpdateResult data =
+    { sweetPollModel : Model data
+    , currentData : Maybe data
+    , error : Maybe Http.Error
```

Comment:
💭 I wonder if this should contain some information on the relative age of `currentData` and `error`. It might be interesting to know whether `currentData` has been updated more recently than `error`, or whether it's the other way around.

Alternatively, if keeping track of that type of information is the considered responsibility of the library user, I think we could communicate that more clearly by passing a state diff here (something like `change : NoChange | NewData data | NewError Http.Error`).